File: Server/Server.Api/Entities/Survey.cs
using System.Collections;
using System;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;

namespace Server.Api.Entities
{
    public record Survey
    
    {
        public int Id { get; init; }

        [Required]
        public string Name { get; init; }

        [Required]
        public DateTimeOffset OpeningDate { get; init; }

        [Required]
        public DateTimeOffset ClosingDate { get; init; }

        [Required]
        public DateTimeOffset CreatedDate { get; init; }

        [Required]
        public ICollection<Question> Questions { get; init; }

        [Required]
        public ICollection<Team> Teams { get; init; }

        [Required]
        public Iteration Iteration { get; init; }

        [Required]
        public int IterationId { get; init; }

        public ICollection<SurveyAttempt> Attempts { get; init;}

    }
}

File: Client/src/models/survey.ts
import Question from "./question"
import Team from "./team"
//Interface for a single survey, and optionally it's questions, and teams it has been distributed to.
export default interface Survey {
  id: number
  name: string
  openingDate: Date
  closingDate: Date
  questions?: Question[]
  teams?: Team[]
}
